Rename `TSDataset.index` -> `TSDataset.timestamps`, `TSDataset.add_columns_from_pandas` -> `TSDataset.add_features_from_pandas`, `TSDataset.update_columns_from_pandas` -> `TSDataset.update_features_from_pandas`

ArtemLiA opened the pull request:
## Before submitting (must do checklist)

- [x] Did you read the [contribution guide](https://github.com/etna-team/etna/blob/master/CONTRIBUTING.md)?
- [ ] Did you update the docs? We use Numpy format for all the methods and classes.
- [x] Did you write any new necessary tests?
- [x] Did you update the [CHANGELOG](https://github.com/etna-team/etna/blob/master/CHANGELOG.md)?
<!-- For CHANGELOG separate each item in unreleased section by blank line to reduce collisions -->

## Proposed Changes

Rename `TSDataset.index` to `TSDataset.timestamps` and make it read-only:
- Rename `TSDataset.index` to `TSDataset.timestamps`
- Make it return `self.df.index.copy()` (make read-only)
- Fix all places where `ts.index` was used and replace them with `ts.timestamps`

Rename `TSDataset.add_columns_from_pandas` to `TSDataset.add_features_from_pandas`:
- Rename `TSDataset.add_columns_from_pandas` to `TSDataset.add_features_from_pandas`
- Replace all usage of old name
- Fix tests

Rename `TSDataset.update_columns_from_pandas` to `TSDataset.update_features_from_pandas`:
- Rename `TSDataset.update_columns_from_pandas` to `TSDataset.update_features_from_pandas`
- Replace all usage of old name
- Fix tests and fixtures
 

## Closing issues
Closes #564 #567

d-a-bunin commented on CHANGELOG.md:
```diff
@@ -36,9 +36,9 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
-- 
-- 
-- 
+- Rename `TSDataset.index` to `TSDataset.timestamps` ([#593](https://github.com/etna-team/etna/pull/593))
```
Add prefix: `**Breaking:**`, because these are breaking changes.

d-a-bunin commented:
Hi! It is probaby the same problem as in #563. Look at the comments. The probable solution can be seen [here](https://github.com/etna-team/etna/pull/570/files#diff-799d73d5834fb4c1c451d8abc5e9fb2b50206058a354cfe6be9039a19ab2cc41).

ArtemLiA commented:
Thanks! I found the reason. Error was in typing annotation of `forecasts_ts` fixtures. It returns and should return `List[pd.DataFrame]` instead of `List[TSDataset]`.

I fixed annotation and renamed `forecasts_ts` to `forecasts_df` so that name doesn't mislead anyone in the future.

d-a-bunin reviewed (approved):
Looks good!
